File: 876 Middle Node/Main.java
class Main {
    public static void main(String[] args) {
        
        Node node1 = new Node(1);
        Node node2 = new Node(2);
        Node node3 = new Node(3);
        Node node4 = new Node(4);
        Node node5 = new Node(5);

        node1.next = node2;
        node2.next = node3;
        node3.next = node4;
        node4.next = node5;
        
        //Storing the res;
        Node res = middleNoed(node1);

        // Traverse from the middle node!!
        traverse(res);

    }

    private static void traverse(Node head ) {
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + "->");
            temp = temp.next;
        }

        System.out.println("null");
    }

    private static Node middleNoed(Node head) {
        Node slow = head;
        Node fast = head;

        while(fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }

        return slow;
    }
}

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

File: 876 Middle Node/main.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class Main:
    def traverse(self, head):
        temp = head
        while temp is not None:
            print(temp.data, end="->")
            temp = temp.next
        print("null")

    def middelNode(self, head):
        slow = head
        fast = head

        while fast is not None and fast.next is not None:
            slow = slow.next
            fast = fast.next.next
        
        return slow

node1 = Node(1)
node2 = Node(2)
node3 = Node(3)
node4 = Node(4)
node5 = Node(5)

node1.next = node2
node2.next = node3
node3.next = node4
node4.next = node5

main = Main()
res = main.middelNode(node1)
main.traverse(res)